File: 1890-sum-of-beauty-of-all-substrings/1890-sum-of-beauty-of-all-substrings.cpp
class Solution {
public:
    int beautySum(string s) {
        int totalBeauty = 0;
        
        for (int i = 0; i < s.size(); i++) {
            vector<int> freq(26, 0);  
            
            for (int j = i; j < s.size(); j++) {
                freq[s[j] - 'a']++;  
                
                int maxi = 0, mini = INT_MAX;
                for (int f : freq) {
                    if (f > 0) {
                        maxi = max(maxi, f);
                        mini = min(mini, f);
                    }
                }
                
                totalBeauty += (maxi - mini);
            }
        }
        
        return totalBeauty;
    }
};

File: 1890-sum-of-beauty-of-all-substrings/1890-sum-of-beauty-of-all-substrings.java
class Solution {
  
    public int beautySum(String s) {
        int ans = 0;
        for(int i=0;i<s.length();i++){
              int[] freq = new int[26];
              Arrays.fill(freq,0);
            for(int j=i;j<s.length();j++){
               freq[s.charAt(j) - 'a']++;
                int mini = Integer.MAX_VALUE, maxi = Integer.MIN_VALUE;
               for(int k=0;k<26;k++){
                 if (freq[k] > 0) { // only consider characters present
                        maxi = Math.max(maxi, freq[k]);
                        mini = Math.min(mini, freq[k]);
                    }
               }
              
            
          
  
            ans += (maxi - mini);
            }
        }
        return ans;
    }
}